### PR TITLE
Use StreamConsumer from gstreamer itself

### PR DESCRIPTION
**Pull request by thiblahute**
Merged in https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/merge_requests/1022

### CODE REVIEW

File: plugins/src/webrtcsink/imp.rs [hunk 1]
```diff
@@ -1257,11 +1269,9 @@ impl Consumer {
...
         }
 
-        let appsrc = appsrc.downcast::<gst_app::AppSrc>().unwrap();
-
-        appsrc.set_format(gst::Format::Time);
-        appsrc.set_is_live(true);
-        appsrc.set_handle_segment_change(true);
+        self.src.set_format(gst::Format::Time);
+        self.src.set_is_live(true);
+        self.src.set_handle_segment_change(true);
```
[the remark this answers is not in the document]
Comment (thiblahute) [reply plugins/src/webrtcsink/imp.rs:1274]: We can simply remove those lines indeed.

[the remark this answers is not in the document]
Comment (MathieuDuponchelle) [reply plugins/src/webrtcsink/imp.rs:1274]: There's a trap here, iirc you need to call these *before* the state of the pipeline is brought up from NULL

[the remark this answers is not in the document]
Comment (sdroege) [reply plugins/src/webrtcsink/imp.rs:1274]: That's exactly why I added that new function in the MR and linked to it here ;)

[the remark this answers is not in the document]
Comment (MathieuDuponchelle) [reply plugins/src/webrtcsink/imp.rs:1274]: ack, jusat saying you can't "just remove these lines" :)

[the remark this answers is not in the document]
Comment (thiblahute) [reply plugins/src/webrtcsink/imp.rs]: Yep, got to call the method there instead.

[the remark this answers is not in the document]
Comment (thiblahute) [reply plugins/src/webrtcsink/imp.rs]: Done.

**Comment by MathieuDuponchelle** [comment]
Let's get this in, thanks @thiblahute very cool :)

**Pull request merged**